### DISCUSSION TITLE
ALTER UPDATE vs ReplacingMergeTree

### DISCUSSION THREAD

**Discussion by vlourme** [ClickHouse/ClickHouse#37074] [Q&A] [answered]
Hello,

We're having a debate with my colleagues about updating the data.
I know that ClickHouse is not optimized for data updating, however, our data pipeline has a “data enrichment” step (that is mostly renaming some values). These updates are meant to be run up to 4 times per day.

One of the easiest way of doing this data enrichment step is by updating the data in ClickHouse, for this, we found two ways of doing:
- Doing a simple `ALTER TABLE [table] UPDATE field='new_value' WHERE field='old_value'`
- Using a `ReplicatedMergeTree`, inserting a new row with same ID but new value and then run the `OPTIMIZE`.

We are also thinking of doing this outside of ClickHouse, for example, mapping over results and replacing values from our backend. This requires more work and may not grant us the best performances.

What are best practices about update the data? Should I use `ReplicatedMergeTree` over `ALTER UPDATE`?

**Comment by kokizzu** [comment]
so, one year later, have you found the answer?

**Comment by den-crane** [comment, marked as answer]
ReplicatedMergeTree -> ReplacingMergeTree

>What are best practices about update the data? Should I use ReplicatedMergeTree over ALTER UPDATE?

They both are great but for the different use cases. 

`ALTER TABLE [table] UPDATE` is for rare updates like one update per date. Also this operation is asynchronous and can take time, hours, days.
